test/server/server_test.cc

```diff
@@ -423,6 +440,37 @@ TEST_P(ServerInstanceImplTest, LifecycleNotifications) {
   server_thread->join();
 }
 
+TEST_P(ServerInstanceImplTest, DrainParentListenerAfterWorkersStarted) {
```

Comment:
Can you run this with `--runs_per_test=1000` to make sure it doesn't flake?

Reply:
Done.

```
 Target //test/server:server_test up-to-date:
  bazel-bin/test/server/server_test
INFO: Elapsed time: 4724.860s, Critical Path: 22.74s
INFO: 1001 processes: 1 internal, 1000 processwrapper-sandbox.
INFO: Build completed successfully, 1001 total actions
//test/server:server_test                                                PASSED in 21.8s
  Stats over 1000 runs: max = 21.8s, min = 18.5s, avg = 18.8s, dev = 0.3s
```